tests/unit/handlers/__init__.py

```diff
@@ -11,3 +11,39 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+
+# Utility functions to setup mock OpenTelemetry spans, needed by multiple test
+# suites.
+
+import contextlib
+import mock
+
+import opentelemetry.context
+import opentelemetry.trace
+
+from opentelemetry.trace.span import TraceFlags
+
+_OTEL_SPAN_CONTEXT_TRACE_ID = 0x123456789123456789
+_OTEL_SPAN_CONTEXT_SPAN_ID = 0x123456789
+_OTEL_SPAN_CONTEXT_TRACEFLAGS = TraceFlags(TraceFlags.SAMPLED)
+
+_EXPECTED_OTEL_TRACE_ID = "00000000000000123456789123456789"
+_EXPECTED_OTEL_SPAN_ID = "0000000123456789"
+_EXPECTED_OTEL_TRACESAMPLED = True
+
+
+@contextlib.contextmanager
+def _setup_otel_span_context():
+    """Sets up a nonrecording OpenTelemetry span with a mock span context that gets returned
+    by opentelemetry.trace.get_current_span, and returns it as a contextmanager
+    """
+    span_context = opentelemetry.trace.SpanContext(
+        _OTEL_SPAN_CONTEXT_TRACE_ID,
+        _OTEL_SPAN_CONTEXT_SPAN_ID,
+        False,
+        trace_flags=_OTEL_SPAN_CONTEXT_TRACEFLAGS,
+    )
+    span = opentelemetry.trace.NonRecordingSpan(span_context)
+    with mock.patch("opentelemetry.trace.get_current_span", return_value=span) as m:
+        yield m
```

Comment:
You don't actually need to do any mocking by just setting the span in the real context implementation, see this example https://opentelemetry.io/docs/languages/python/cookbook/#manually-setting-span-context:~:text=%23%20Or%20you%20can,(token)
```suggestion
    ctx = trace.set_span_in_context(span)
    token = context.attach(ctx)
    try:
        yield
    finally:
        context.detach(token)
```

I am guessing that's why you are using `import opentelemetry.trace` instead of `from opentelemetry import trace`